apps/studio/components/interfaces/Auth/AuthProvidersFormValidation.tsx

```diff
@@ -843,78 +844,32 @@ const EXTERNAL_PROVIDER_GOOGLE = {
-      .when(['EXTERNAL_GOOGLE_ENABLED', 'EXTERNAL_GOOGLE_CLIENT_ID'], {
-        is: (EXTERNAL_GOOGLE_ENABLED: boolean, EXTERNAL_GOOGLE_CLIENT_ID: string) => {
-          return EXTERNAL_GOOGLE_ENABLED && !!EXTERNAL_GOOGLE_CLIENT_ID
-        },
-        then: (schema) =>
-          schema
-            .matches(
-              /^[a-z0-9.\/_-]*$/i,
-              'Invalid characters. Google OAuth Client Secrets usually contain letters, numbers, dots, dashes and underscores.'
-            )
-            .required('Client Secret is required when using the OAuth flow.'),
-      })
-      .when(
-        [
-          'EXTERNAL_GOOGLE_ENABLED',
-          'EXTERNAL_GOOGLE_ADDITIONAL_CLIENT_IDS',
-          'EXTERNAL_GOOGLE_CLIENT_ID',
-        ],
-        {
-          is: (
-            EXTERNAL_GOOGLE_ENABLED: boolean,
-            EXTERNAL_GOOGLE_ADDITIONAL_CLIENT_IDS: string,
-            EXTERNAL_GOOGLE_CLIENT_ID: string
-          ) => {
-            return (
-              EXTERNAL_GOOGLE_ENABLED &&
-              !!EXTERNAL_GOOGLE_ADDITIONAL_CLIENT_IDS &&
-              !EXTERNAL_GOOGLE_CLIENT_ID
-            )
-          },
-          then: (schema) =>
-            schema.matches(
-              /^$/,
-              'Client Secret should only be set when Client ID for OAuth is set.'
-            ),
-        }
-      ),
-    EXTERNAL_GOOGLE_CLIENT_ID: string().matches(
-      /^([a-z0-9-]+([.][a-z0-9-]+)+)?$/i,
-      'Invalid characters. Google OAuth Client IDs are usually a domain-name (e.g. 01234567890-abcdefghijklmnopqrstuvwxyz012345.apps.googleusercontent.com).'
-    ),
-    EXTERNAL_GOOGLE_ADDITIONAL_CLIENT_IDS: string()
+    EXTERNAL_GOOGLE_CLIENT_ID: string()
+      .matches(/^\S+$/, 'Client IDs should not contain spaces.')
       .matches(
         /^([a-z0-9-]+([.][a-z0-9-]+)*(,\s*[a-z0-9-]+([.][a-z0-9-]+)*)*,*\s*)?$/i,
-        'Invalid characters. Google Client IDs are usually a domain-name style string (e.g. com.example.com.app or *.apps.googleusercontent.com).'
+        'Invalid characters. Google Client IDs should be a comma-separated list of domain-like strings.'
       )
-
-      .when(['EXTERNAL_GOOGLE_ENABLED', 'EXTERNAL_GOOGLE_CLIENT_ID'], {
-        is: (EXTERNAL_GOOGLE_ENABLED: boolean, EXTERNAL_GOOGLE_CLIENT_ID: string) => {
-          return EXTERNAL_GOOGLE_ENABLED && !EXTERNAL_GOOGLE_CLIENT_ID
-        },
-        then: (schema) =>
-          schema.required(
-            'At least one Authorized Client ID is required when not using the OAuth flow.'
+      .required('At least one Client ID is required when Google sign-in is enabled.'),
+    EXTERNAL_GOOGLE_SECRET: string().when('EXTERNAL_GOOGLE_ENABLED', {
+      is: true,
+      then: (schema) =>
+        schema
+          .required('Client Secret is required when using the OAuth flow.')
+          .matches(
+            /^[a-z0-9.\/_-]*$/i,
+            'Invalid characters. Google OAuth Client Secrets usually contain letters, numbers, dots, dashes, and underscores.'
```

Comment:
does this handle the case where one should be allowed to add a client id without a client secret?

Reply:
Just tested it, and it does. It treats the client secret as an optional field. 

<img width="765" alt="image" src="https://github.com/user-attachments/assets/c94a51fc-4056-4b22-a848-fe6e3e645c98">

Reply:
The Secret Key is only required if Google or Apple provider is enabled.

Reply:
hmm but if the provider is not enabled, how do they edit those fields? I was under the impression that you can't edit the client ID and secret fields without first enabling the provider? 

also, what happens if the provider is enabled? will it allow the user to add only the client id field?

Reply:
@kangmingtay I didn’t make changes to this functionality. If you check in production, users can currently edit the Client ID and Secret fields even when the provider isn’t enabled. 

But if the provider is enabled, users will be required to add at least one Client ID and a Secret now. Should I set it up so that only the Client ID is required?

Reply:
> Should I set it up so that only the Client ID is required?
yup this should be allowed - can you also check if it allows saving only the client ID when the provider is enabled? if it's doesn't allow that, i can help to make the corresponding changes in the backend too

Reply:
might be a mix up here, as the whole form with all values get submitted on each submit.
So 'just changing the ID' will mean all values are submitted.

Reply:
in other words, all required fields need to have a value if 'enabled' is true.
you can however just change 1 input and fire the form, but technically your submitting all values.